test/tests/functional/pbs_multi_sched.py

```diff
@@ -1322,3 +1322,50 @@ def test_list_multi_sched(self):
             err_msg = "Unknown Scheduler"
             self.assertTrue(err_msg in e.msg[0],
                             "Error message is not expected")
+
+    def test_multisched_sched_cycle_length(self):
+        """
+        Test the scheduler attribute sched_cycle_length for multisched
+        """
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'False'},
+                            expect=True)
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'sched_cycle_length': '00:00:01'},
+                            id='default',
+                            expect=True)
+        a = {'resources_available.ncpus': 100}
+        self.server.manager(MGR_CMD_SET, NODE, a, self.mom.shortname)
+        for _ in xrange(100):
+            j = Job(TEST_USER)
+            self.server.submit(j)
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'},
+                            expect=True)
+        t = int(time.time())
+        msg = ";Leaving the scheduling cycle:" +\
+              " Cycle duration of 1 seconds has exceeded" +\
+              " sched_cycle_length of 1 seconds"
+        self.scheduler.log_match(msg, starttime=t)
+        self.setup_sc2()
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'scheduling': 'False'}, id="sc2")
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'sched_cycle_length': '00:00:02'},
+                            id='sc2', expect=True)
+        a = {'queue_type': 'execution', 'started': 'True',
+             'enabled': 'True'}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id='wq1')
+        p2 = {'partition': 'P2'}
+        self.server.manager(MGR_CMD_SET, QUEUE, p2, id='wq1', expect=True)
+        a = {'resources_available.ncpus': 250, 'partition': 'P2'}
+        self.server.create_vnodes('vnode', a, 1, self.mom)
+        self.server.manager(MGR_CMD_SET, NODE, p2, id='vnode[0]', expect=True)
+        for _ in xrange(250):
+            j = Job(TEST_USER, attrs={ATTR_queue: 'wq1'})
+            self.server.submit(j)
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'scheduling': 'True'}, id="sc2", expect=True)
+        t = int(time.time())
+        msg = ";Leaving the scheduling cycle:" +\
+              " Cycle duration of 2 seconds has exceeded" +\
+              " sched_cycle_length of 2 seconds"
+        self.scheds['sc2'].log_match(msg, starttime=t)
```

Comment:
We have seen in other tests that the jobs submitted by one test affects the run time of the next test because setUp of the next test takes a lot of time to delete this job.
I'd suggest that you delete all the jobs you are submitting within this test itself.
If you choose to go server dynamic resource route then you do not have this problem at all.

Reply:
Added code to clean all the jobs at the end.